xz/FeatureModel.xml

```diff
@@ -741,17 +741,6 @@
             <column>34</column>
           </end>
         </sourceRange>
-        <sourceRange>
```

Comment:
Well, this one is problematic. Currently, the locations in this model are base on commit `4773608554d1b684a05ff9c1d879cf5c42266d33` and in this one, the locations is correct.

We need to find a way, to correctly hand his :thinking: 
@ChristianKaltenecker @Kallistos Ideas?

Reply:
The problem here is that I'm not sure if the respective variable (named `init_format`) should be considered as a feature variable. During command line parsing, the variable at the other location is set (named `opt_format`), so this one is definitely a feature variable of the feature. `init_format` is then just used as a local helper variable of `opt_format`.

Reply:
@ChristianKaltenecker What is your point on this? What did we do in the past for `xz`? Could you please double check this.

Reply:
1. We could introduce a certain attribute to the feature models containing the commit hash or release this feature model relates to.
2. I had a look on this and agree with @Keller-1998 . From a black-box perspective it shouldn't really make a difference. In the past, we ignored such helper variables. If anything changes in the performance behavior of xz by changing the `opt_format` (even if helper variables are used internally), then it is attributed to `opt_format`. This is also fine from the user perspective since a user usually doesn't know about some internal helper variables.

Reply:
So, we then need a revision range for a feature.
```
<RevisionRange>
    <introduced=C_HASH>
    <removed=C_HASH>
</RevisionRange>
```

@s9latimm What do you think?

Reply:
Instead of a feature attribute, we could add this `RevisionRange` to a `sourceRange`. 
```
<locations>
  <sourceRange>
    <RevisionRange>
      <start>...</start>
      <end>...</end>
    </RevisionRange>
    ...
  </sourceRange>
  ...
</locations>
```
As we already support multiple locations for a feature, this would work to some degree out of the box.

Reply:
We can still define it on the sourceRange level and have it "implicit" on the feature.
The feature RevisionRange would then be the set off all revisions where at least on sourceRange is valid.

Reply:
> Instead of a feature attribute, we could add this `revisionRange` to a `sourceRange`.
> 
> ```
> <locations>
>   <sourceRange>
>     <revisionRange>
>       <introduced>...</introduced>
>       <removed>...</removed>
>     </revisionRange>
>     ...
>   </sourceRange>
>   ...
> </locations>
> ```

Will be added with https://github.com/se-sic/vara-feature/pull/101.